preact.d.ts

```diff
@@ -0,0 +1,4 @@
+declare module "preact/compat/server" {
```

Comment:
The `@preact/compat` alias is doing a bad job here since Next.js now uses ESM imports, so I just made those imports use `preact/compat` directly. I know it defeats the meaning of `@preact/compat` somehow but we _are_ using Preact anyway so it's not a big deal.